git_machete/cli.py

```diff
@@ -360,21 +360,21 @@ def update_cli_opts_using_parsed_args(
         parsed_args: argparse.Namespace) -> None:
     for opt, arg in vars(parsed_args).items():
         if opt == "branch":
-            cli_opts.opt_branch = arg
```

Comment:
Okay... and if it comes to the original false positive in mypy (i.e. no error shown even though it should)... is there any flag (in mypy.ini) that could be switched on to prevent further errors like this one? or is it just a bug in mypy itself? maybe it should be reported in https://github.com/python/mypy/issues if so 🤔

Reply:
Huh, it took me quite a long to time to dig into mypy so deep. Unfortunately, this is normal mypy behaviour. Arguments that come from untyped functions/variables are silently treated by mypy as `Any`. Since `argparse` is not typed, everything that comes from `argparse.Namespace` will be taken as `Any` :(  
Even if we add `type=LocalBranchShortName` into argument parser for `branch`, python debugger will see `branch` as `LocalBranchShortName` but mypy always will see it as `Any` until you specifically tell mypy what is the exact type by casting (right now it's done this way, but casting does not solve the problem). There is no such a flag to change it. 

I also tried to build a class after `argparse.Namespace` with desired options typed, but still mypy sees them as `Any`.

The only reasonable solution here would be to use [TAP](https://pypi.org/project/typed-argument-parser/) which is a wrapper over `argparse`  with modernised solution for typing. 
But it would add external dependency to git-machete.

Personally, i would stick to precise casting rather than adding TAP. TBD

Reply:
Oh! okay, I see now ☹️ thanks for the breakdown!

Still, given the amount of instability that adding external deps will probably add... let's stick to the current approach!

Reply:
Actually! the only thing that could be done on this PR before merge is to add some comment in code that would specifically warn the future generations of this caveat (roughly your GH comment from above, maybe a bit redacted). Otherwise approving, thx for the research again